hypothesis-python/src/hypothesis/internal/conjecture/shrinker.py

```diff
@@ -211,8 +275,10 @@ def clear_passes(self):
 
         self.known_programs = set()
 
-    def add_new_pass(self, run, classification=PassClassification.CANDIDATE):
+    def add_new_pass(self, run, classification=None):
         """Creates a shrink pass corresponding to calling ``run(self)``"""
+        if classification is None:
+            classification = PassClassification.CANDIDATE
```

Comment:
Is there any particular motivation behind this change?

Reply:
Stupid Python shenanigans. I wanted the `Shrinker` class at the top because it's the main entry point, but this means that `PassClassification` definition is not in scope at the time when the defaults are defined.

Reply:
IMO it's more elegant to just define the enum first, but YMMV I guess!